lab_7_llm/main.py

```diff
@@ -55,12 +60,25 @@ def analyze(self) -> dict:
         Returns:
             dict: Dataset key properties
         """
+        rows, cols = self._raw_data.shape
+        data_droped_empty = self._raw_data.dropna()
+
+        return {'dataset_number_of_samples': rows,
+                'dataset_columns': cols,
+                'dataset_duplicates': len(self._raw_data[self._raw_data.duplicated()]),
+                'dataset_empty_rows': rows - len(data_droped_empty),
+                'dataset_sample_min_len': min(data_droped_empty['ru'].str.len()),
+                'dataset_sample_max_len': max(data_droped_empty['ru'].str.len())}
```

Comment:
Use `ColumnNames.SOURCE.value` instead of `ru`

Reply:
I don't understand. This method is called before ```transform```, so the column name is 'ru' not 'source'. Or should I use my own enum for this?

Reply:
I forgot that we are renaming the columns. Then your decision is correct